### ISSUE TITLE
categorie as a plain String

### ISSUE THREAD

**Issue by melocotton** [Harsisis/Java_project#4]
TLDR, for predefined limites values, it is better to use an enum.

A categorie is a predefined value among few values. As such, it should not be possible to put anything else than just the predefined “Support numérique” and the few other categories. The fact that you hardcoded this string in the constructor does not really protect you from having incorrect values in the future : imagine that you need a feature where you can change a product from one categorie to another (let say you create a new “manga” categorie, but atm mangas are in “BD”). You will certainly end up creating a setCategorie(...) with a string as input Param... then you will have to either check the input value and reject any invalid string in this setter (but then you still need to know what are the acceptable values ?), or you can control the parameter before you call the setter.. but then this is even worse because you have to check this everywhere you use it : code duplication, etc.
